packages/client/ui/react-ui/src/components/auth/PasskeyPrompt.tsx

```diff
@@ -0,0 +1,266 @@
+                        "fixed inset-0 bg-[rgba(139,151,151,0.2)] backdrop-blur-[2px] transition-opacity duration-300 ease-in-out -z-10"
+                    )}
+                />
+            </Transition.Child>
+            <div
+                className={classNames(
+                    "flex flex-col items-center w-full max-w-[28rem] rounded-2xl shadow-sm z-30 border",
+                    appearance?.colors?.background != null ? `bg-[${appearance.colors.background}]` : "bg-white",
+                    appearance?.colors?.border != null ? `border-[${appearance.colors.border}]` : "border-[#D0D5DD]"
+                )}
+                onClick={(e) => e.stopPropagation()}
+            >
+                <div className="pt-12 pb-10 px-8">
+                    <div className="flex justify-center left-1.5 relative">
+                        <PasskeyPromptLogo appearance={appearance} />
+                    </div>
+
+                    <div className="flex justify-center">
+                        <p
+                            className={classNames(
+                                "text-lg font-bold",
+
+                                appearance?.colors?.textPrimary != null
+                                    ? `text-[${appearance.colors.textPrimary}]`
+                                    : "text-[#20343E]"
+                            )}
+                        >
+                            {title}
+                        </p>
+                    </div>
+
+                    <div className="mt-4 mb-9">
+                        <div
+                            className={classNames(
+                                "font-normal",
+                                appearance?.colors?.textSecondary != null
+                                    ? `text-[${appearance.colors.textSecondary}]`
+                                    : "text-[#67797F]"
+                            )}
+                        >
+                            {content}
+                        </div>
+                    </div>
+
+                    <div className="flex flex-col gap-4 justify-center">
+                        {primaryButton}
+                        {secondaryAction}
+                    </div>
+                    <div className="flex justify-center pt-4">
+                        <PoweredByCrossmint color={appearance?.colors?.textSecondary} />
+                    </div>
+                </div>
+            </div>
+        </Dialog>
+    );
+}
+
+type PromptType = "create-wallet" | "transaction" | "not-supported" | "create-wallet-error" | "transaction-error";
+
+export function PasskeyPrompt({
+    state,
+    appearance,
+}: {
+    state: {
+        type: PromptType;
+        primaryActionOnClick: () => void;
+        secondaryActionOnClick?: () => void;
+    };
+    appearance?: UIConfig;
+}) {
+    // These components are currently assembled based on the mockups.
+    if (state.type === "create-wallet") {
```

Comment:
this function is pretty long and hard to read, 2 thoughts:
1. the branching logic is always on `state.type`, so a `switch` is more appropriate 
2. Each returned component is a customized instance of `PasskeyPromptCore`, but they each require 10-20 lines of customization. 

WHat do you think of this?
```
type PasskeyPromptProps = {
  state: {
        type: PromptType;
        primaryActionOnClick: () => void;
        secondaryActionOnClick?: () => void;
    };
    appearance?: UIConfig;
}
export function PasskeyPrompt({ state, appearance}: PasskeyPromptProps) { // whenever possible, I like to keep function declarations to one line, IMO much easier to read 
  switch (state.type) {
    case "create-wallet":
      return <CreateWalletPasskeyPrompt appearance={appearance} state={state}
    case "create-wallet-error":
       return <CreateWalletErrorPrompt appearance={appearance} state={state}
    ...
    default:
      console.error("unexpected case blah ", state.type"
}

function CreateWalletPasskeyPrompt(state, appearance) { 
  return <PasskeyPromptCore
                title="Create Your Wallet"
                appearance={appearance}
                content={
                    <>
                        <div className="mb-4">You're about to create a wallet.</div>
...

Reply:
ah so your thinking of creating explicit prompt type components ie. CreateWalletPasskeyPrompt, FirstSignTransactionPrompt etc..?

Reply:
I've updated to use switch statements 👍

Reply:
i've also updated the type for the one-line function definition 👍